src/Orleans.Serialization/Codecs/IPAddressCodec.cs

```diff
@@ -83,23 +86,15 @@ void IFieldCodec<IPAddress>.WriteField<TBufferWriter>(ref Buffers.Writer<TBuffer
             }
 
             writer.WriteFieldHeader(fieldIdDelta, expectedType, CodecFieldType, WireType.LengthPrefixed);
-#if NET5_0_OR_GREATER
-            Span<byte> buffer = stackalloc byte[64];
-            if (value.TryWriteBytes(buffer, out var length))
-            {
-                var writable = writer.WritableSpan;
-                if (writable.Length > length)
-                {
-                    writer.WriteVarUInt32((uint)length);
-                    buffer.Slice(0, length).CopyTo(writable.Slice(1));
-                    writer.AdvanceSpan(length);
-                    return;
-                }
-            }
-#endif
-            var bytes = value.GetAddressBytes();
-            writer.WriteVarUInt32((uint)bytes.Length);
-            writer.Write(bytes);
+
+            Unsafe.SkipInit(out Guid tmp);
+            var buffer = MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref tmp, 1));
+            value.TryWriteBytes(buffer, out var length);
+            Debug.Assert(length > 0);
+            buffer = buffer[..length];
+
+            writer.WriteVarUInt32((uint)buffer.Length);
+            writer.Write(buffer);
```

Comment:
Is the idea here that we will fix this code if/when IPAddress becomes longer than 16 bytes?
If so, it seems that the `Debug.Assert` should be a non-conditional check+throw (since this check will not be emitted in Release builds).
It seems fragile not to have an alternative path in case the optimization does not apply.

Unless there's a guarantee somewhere that `IPAddress` will never be longer than that.

Reply:
I think this kind of code also warrants comments, since it's being a little tricky: allocating a `Guid` worth of uninitialized space on the stack and using it as a span of bytes. I wonder if there are other places a trick like this might apply.

Reply:
I changed the assert to a check+throw. It's unlikely that `IPAddress` will grow larger than IPv6 anytime soon.

Also added a comment around `Unsafe.SkipInit` - it's a limitation of C#10 that assumes that the `Span<byte> buffer` (a ref-like type) could escape this function since `writer` itself is a ref parameter and a `ref struct`. C#11 will add `scoped` ref parameter support for exactly such cases.

Reply:
Sounds great to me. I'm learning from your optimizations, too.